### DISCUSSION TITLE
Pixel with Snapdragon compatibility

### DISCUSSION THREAD

**Discussion by mightyvenom** [kyujin-cho/pixel-volte-patch#5] [Q&A] [answered]
Hi, do you know if this mod can safely be tested on Pixel 4a/5 devices (with snapdragon)?

Do you think it can work?

Thank you!

**Comment by kyujin-cho** [comment, marked as answer]
Hi. AFAIK enabling VoLTE manually on Pixel 4a/5 requires modifying EFS partition, so this won't work.

**Comment by MaticSulc** [comment]
I have tested it on Pixel 5, as expected - it does not work, but it doesn't break anything.

**Comment by mightyvenom** [comment]
@kyujin-cho I realize that getting IMS registered for qualcomm pixels requires modyfing the EFS partition but I am trying to figure out if we can use Shizuku to call some API that might force IMS registration? I am looking at the docs listed here https://developer.android.com/reference/android/telephony/CarrierConfigManager.Ims but since I am not that technical I can't tell if there is something that can be done here.

Maybe modifying [KEY_IMS_USER_AGENT_STRING](https://developer.android.com/reference/android/telephony/CarrierConfigManager.Ims#KEY_IMS_USER_AGENT_STRING) to trick the system we are using a different phone (eg. Nokia 8.3 5G which also uses the same Snapdragon 765G) will work?

OR [KEY_IMS_SINGLE_REGISTRATION_REQUIRED_BOOL](https://developer.android.com/reference/android/telephony/CarrierConfigManager.Ims#KEY_IMS_SINGLE_REGISTRATION_REQUIRED_BOOL) might do the trick.

FYI the required modem mbn configs for qualcomm pixels are located here: /system/vendor/rfs/msm/mpss/readonly/vendor/mbn (more info [here](https://github.com/stanislawrogasik/Pixel5-VoLTE-VoWiFi/tree/main/system/vendor/rfs/msm/mpss/readonly/vendor/mbn/mcfg_sw))

Can you can have a look to see if any of the IMS strings can be called from there using Shizuku and thus getting the device registered?

As I have posted on XDA forums [here](https://forum.xda-developers.com/t/mod-magiskmodule-volte-vowifi-5g-in-unsupported-countries-pixel-4a-5g-5-5a.4547909/post-88262319), using your module enables the VOLTE/VoWiFi switches but they will not work since IMS remains unregistered.

**Reply by kyujin-cho** [reply]
- In my experience altering IMS User-Agent string is mostly affected when trying to establish IMS connections (e.g. calling or receiving call), not when registering device to carrier's IMS server.
- As you mentioned it seems that enabling VoLTE on the phone requires at least matching profile offered by carrier (I'm not an expert about the cellular connections so I don't know how the engineers actually call it). It seems like injecting MBN files is equivalent to adding those profiles manually to device. Unfortunately there's no known system API to alter NV partition, and since the shizuku trick only works when a known System API exists.
- Also I haven't seen any kind of system API which sets the IMS profile on-demand (rather than reading from NVRAM).

So, in short: For now, no.

**Comment by mightyvenom** [comment]
Hello @kyujin-cho , now that Snapdragon officially announced its new "Android on Snapdragon" API portal (that was previously vendor locked) can something similar to Tensor Pixels be done for Snapdragon Pixels?

More info [here](https://twitter.com/MishaalRahman/status/1668687202346213395?t=knYu9xSPvYUHjV0tBKCJMg&s=19) 

Would be great to know, thanks!

**Reply by kyujin-cho** [reply]
After reading the [API documentation page](https://www.qualcomm.com/products/mobile/snapdragon/smartphones/android-on-snapdragon), unfortunately, I could not find any kind of API related about cellular modems. So in short, still no.

**Comment by mightyvenom** [comment]
Comment just to re-open discussion
